controllers/authorino_controller.go

```diff
@@ -0,0 +1,598 @@
+//+kubebuilder:rbac:groups=authorino-operator.kuadrant.3scale.net,resources=authorinoes/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=authorino-operator.kuadrant.3scale.net,resources=authorinoes/finalizers,verbs=update
+
+// +kubebuilder:rbac:groups="apps",resources=deployments,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Authorino object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
+func (r *AuthorinoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("authorinoReconciler", req.NamespacedName)
+
+	//get operator namespace
+	r.OperatorNamespace = authorinooperatorv1beta1.AuthorinoOperatorNamespace
+	ns, found := os.LookupEnv("WATCH_NAMESPACE")
+	if found {
+		r.OperatorNamespace = ns
+	}
+
+	// get authorino instance
+	authorinoInstance, err := r.authorinoInstance(req.NamespacedName)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if authorinoInstance == nil {
+		log.Info("Authorino CR not found. returning the reconciler")
+		return ctrl.Result{}, nil
+	}
+
+	log.Info("Found an instance of authorino", "authorinoInstanceName", authorinoInstance.Name)
+	authorinoInstance.Status.Ready = false
+
+	err = r.authorinoServices(authorinoInstance)
+	if err != nil {
+		authorinoInstance.Status.Ready = false
+		log.Error(err, "Failed to create authorino services")
+		return ctrl.Result{}, err
+	}
+
+	err = r.authorinoPermission(authorinoInstance, req.NamespacedName.Namespace)
+	if err != nil {
+		authorinoInstance.Status.Ready = false
+		log.Error(err, "Failed to create authorino permission")
+		return ctrl.Result{}, err
+	}
+
+	existingDeployment := &appsv1.Deployment{}
+	err = r.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      authorinoInstance.GetName(),
+			Namespace: authorinoInstance.GetNamespace(),
+		}, existingDeployment)
+	if err != nil && errors.IsNotFound(err) {
+		newDeployment := r.authorinoDeployment(authorinoInstance)
+		err = r.Create(ctx, newDeployment)
+		if err != nil {
+			authorinoInstance.Status.Ready = false
+			log.Error(err, "Failed to create Authorino deployment resource", newDeployment.Name, newDeployment.Namespace)
+			return ctrl.Result{}, err
+		}
+		// Deployment created successfully - return and requeue
+		return ctrl.Result{Requeue: true}, nil
+	} else if err != nil {
+		authorinoInstance.Status.Ready = false
+		log.Error(err, "Failed to get Deployment for Authorino")
+		return ctrl.Result{}, err
```

Comment:
Same as https://github.com/Kuadrant/authorino-operator/pull/1/files#r740896089. I think you won't get to `handleStatusUpdate(authorinoInstance)` if you return here.

Reply:
I'll create an issue to better handle the status update based on the conditions conventions https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties